errors.go

```diff
@@ -0,0 +1,35 @@
+package ics
+
+const (
+	MalformedCalendarExpectedVCalendarError  = "malformed calendar; expected a vcalendar"
+	MalformedCalendarExpectedBeginError      = "malformed calendar; expected begin"
+	MalformedCalendarExpectedEndError        = "malformed calendar; expected a end"
+	MalformedCalendarExpectedBeginOrEndError = "malformed calendar; expected begin or end"
+
+	MalformedCalendarUnexpectedEndError             = "malformed calendar; unexpected end"
+	MalformedCalendarBadStateError                  = "malformed calendar; bad state"
+	MalformedCalendarVCalendarNotWhereExpectedError = "malformed calendar; vcalendar not where expected"
```

Comment:
In this case and others I think there there are too many errors. I would prefer to see something like:
```go
ErrorMalformedCalendar = errors.New("malformed calendar")
```
Then it be embellished with detail when it is returned like @arran4 suggested in #98:
```go
return nil, fmt.Errorf("%w: expected begin or end", ErrorMalformedCalendar)
```

Reply:
I'm not bothered by the quantity, but I do think regardless, it should be wrapped in a `ErrorMalformedCalendar` that is a good suggestion either direction. 

I have some things to do I will have to get back to this.

Reply:
(Also not a blocker.)

Reply:
The quantity has practical concerns for me: Say I'm writing a server and choosing a HTTP status code. If the user uploaded an invalid calendar I want to return 400. If something else happened, such as the connection dropped and the io.Reader returned an EOF, then I want to return 500. I now have to check if the error was any one of the many parse errors and hope that this module doesn't make any new ones. I'm happy to drop this thread but want to point out that once you ship all theses errors, you can't take it back without a breaking change.

Reply:
I believe this is possible using `error.Is` sorry I didn't ignore this concern I just failed to follow up on it. I am happy to work through a simple example if you like?